builtin/logical/pkiext/nginx_test.go

```diff
@@ -309,15 +471,77 @@ func RunNginxRootTest(t *testing.T, caKeyType string, caKeyBits int, caUsePSS bo
 		"ip_sans":     "127.0.0.1,::1",
 		"sans":        uniqueHostname + ",localhost,localhost4,localhost6,localhost.localdomain",
 	})
-	requireSuccessNonNilResponse(t, resp, err, "failed to create leaf cert")
+	requireSuccessNonNilResponse(t, resp, err, "failed to create server leaf cert")
 	leafCert := resp.Data["certificate"].(string)
 	leafPrivateKey := resp.Data["private_key"].(string) + "\n"
 	fullChain := leafCert + "\n"
 	for _, cert := range resp.Data["ca_chain"].([]string) {
 		fullChain += cert + "\n"
 	}
 
-	cleanup, host, port, networkName, networkAddr, networkPort := buildNginxContainer(t, fullChain, leafPrivateKey)
+	// Issue a client leaf certificate.
+	resp, err = pki.CBWrite(b, s, "issue/testing", map[string]interface{}{
+		"common_name": "testing.client.dadgarcorp.com",
+	})
+	requireSuccessNonNilResponse(t, resp, err, "failed to create client leaf cert")
+	clientCert := resp.Data["certificate"].(string)
+	clientKey := resp.Data["private_key"].(string) + "\n"
+	clientFullChain := clientCert + "\n" + resp.Data["issuing_ca"].(string) + "\n"
+	clientTrustChain := resp.Data["issuing_ca"].(string) + "\n" + rootCert + "\n"
+	clientCAChain := resp.Data["ca_chain"].([]string)
+
+	// Issue a client leaf cert and revoke it, placing it on the main CRL
+	// via rotation.
+	resp, err = pki.CBWrite(b, s, "issue/testing", map[string]interface{}{
+		"common_name": "revoked-crl.client.dadgarcorp.com",
+	})
+	requireSuccessNonNilResponse(t, resp, err, "failed to create revoked client leaf cert")
+	revokedCert := resp.Data["certificate"].(string)
+	revokedKey := resp.Data["private_key"].(string) + "\n"
+	// revokedFullChain := revokedCert + "\n" + resp.Data["issuing_ca"].(string) + "\n"
+	// revokedTrustChain := resp.Data["issuing_ca"].(string) + "\n" + rootCert + "\n"
+	revokedCAChain := resp.Data["ca_chain"].([]string)
+	_, err = pki.CBWrite(b, s, "revoke", map[string]interface{}{
+		"certificate": revokedCert,
+	})
+	require.NoError(t, err)
+	_, err = pki.CBRead(b, s, "crl/rotate")
+	require.NoError(t, err)
+
+	// Issue a client leaf cert and revoke it, placing it on the delta CRL
+	// via rotation.
+	/*resp, err = pki.CBWrite(b, s, "issue/testing", map[string]interface{}{
```

Comment:
I'm not clear what this code is doing here.

Reply:
We'd like to test the behavior of client cert inclusion on the Delta CRL as well, but nginx doesn't support it presently. So I've commented it out.